tensorrt_llm/_torch/models/modeling_deepseekv3.py

```diff
@@ -514,7 +514,8 @@ def compute_routed_output(self, hidden_states, hidden_states_fp4,
         if self.use_dp and self.mapping.tp_size > 1:
             # FP4 all_gather moves this bf16 allgather in to after topk and fp4 quantization
             # to reduce allreduce BW
-            if disable_fp4_allgather() and not self.experts.enable_alltoall:
+            if disable_fp4_allgather() and not self.experts.can_use_alltoall(
+                    hidden_states_fp4 or hidden_states):
```

Comment:
1. The method `can_use_alltoall` is not a method of the base class. It raises if `self.experts` is a instance of `CutlassFusedMoE`.
2. If hidden_states_fp4 is a Tensor, `hidden_states_fp4 or hidden_states` will raise.